Documentation/custom_effects.md

```diff
@@ -48,6 +48,104 @@ These are some tips for writing or converting effects for use with MonoGame.
+
+Otherwise you may get **"Wrong MGFX file version!"** raised by ReadEffect() in Effect.cs when you do a **content.Load<Effect>**
+[because the version written in to the .xnb file is not a match with the value of MGFXVersion 
+that has been put in the run time].
+
+So if you have this error, the first thing to check is that your content processor is the same up-to-dateness as the version of the runtime you are running your game against.
+
+
+#### Shader model compilation:
+(a) For **OpenGL** versions of MonoGame (eg. Mac, Linus, WindowsGL) the shader model must be **SM 3.0 or lower**
+
+(b) For **DirectX** versions of MonoGame (eg. WinRT, Windows8) the shader model must be **SM 4.0 or higher**.
+
+
+
+In the DirectX case it is typical for people to use the:
+
+`vs_4_0_level_9_1`
+and
+`ps_4_0_level_9_1`
+
+compilation profiles, because these provide backwards compatibility for Shader Model 2 shaders.
+
+
+
+#### Windows8 configuration == DX11 (inside the content processor)
+(a) The "Windows8" build configuration is used internally in the MonoGame content processor code to indicate compilation to target DX11.
+
+Look at the code on line 29 of MGEffectProcessor.cs! 
+
+`options.DX11Profile = platform == MonoGamePlatform.Windows8 ? true : false; `
+
+
+
+The **ContentHelper.cs** file in the ContentProcessors project reads 
+the build configuration name you are using then does this:
+
+`switch (platform.ToUpper())`    
+`{`     
+`case "WINDOWS":`     
+`     return MonoGamePlatform.Windows;`       
+`case "WINDOWS8":`       
+`     return MonoGamePlatform.Windows8;`       
+`case "IOS":`      
+`     return MonoGamePlatform.iOS;`     
+`case "ANDROID":`    
+`     return MonoGamePlatform.Android;`     
+`case "LINUX":`     
+`     return MonoGamePlatform.Linux;`     
+`case "OSX":`     
+`     return MonoGamePlatform.OSX;`     
+`case "PSM":`     
+`     return MonoGamePlatform.PSM;`     
+`etc.`     
+`}`      
+
+Which then gets used by MGEffectProcessor.cs (as above)
+`options.DX11Profile = platform == MonoGamePlatform.Windows8 ? true : false; `
+to set the flag specifying DX11.
+
+In other words, in order to compile .xnb files from your .fx files that will be able to
+run against the Windows DirectX version of MonoGame, you **MUST** use build profile that
+is **specifically called "Windows8" !!**  ... 
+either the one supplied with the project code or one you make yourself.
+
+At some point this getting written into a registry key  
+called "MONOGAME_PLATFORM" (HKEY_CURRENT_USER/Environment/MONOGAME_PLATFORM),
+and then gets read back from that key via:
+`var platform = Environment.GetEnvironmentVariable("MONOGAME_PLATFORM", EnvironmentVariableTarget.User);`
+
+If you don't realize this, having just converted all your .fx files to run against  
```

Comment:
Don't put newlines in the middle of sentences...    markdown is not like HTML... it cares about newlines.  You end up with text that wraps prematurely which looks poor and wastes space.